src/components/TermsModal/index.tsx

```diff
@@ -1,123 +1,130 @@
-  font-weight: 300;
-  flex: 1;
-  font-size: 18px;
-  line-height: 26px;
-  letter-spacing: 0.1px;
-  margin-top: 22px;
+    color: ${({ theme: { colors } }) => colors.primary};
+    font-weight: 300;
+    flex: 1;
+    font-size: 18px;
+    line-height: 26px;
+    letter-spacing: 0.1px;
+    margin-top: 22px;
 `;
 
 const Link = styled.span`
-  color: ${({ theme: { colors } }) => colors.accent};
-  cursor: pointer;
-  font-weight: ${({
-    theme: {
-      typo: { weights }
-    }
-  }) => weights.normal};
+    color: ${({ theme: { colors } }) => colors.accent};
+    cursor: pointer;
+    font-weight: ${({
+        theme: {
+            typo: { weights },
+        },
+    }) => weights.normal};
 `;
 
 const AgreeButton = styled(Button)`
-  width: 194px;
-  height: 50px;
-  padding: 0;
+    width: 194px;
+    height: 50px;
+    padding: 0;
 `;
 
 interface Props {
-  isOpen: boolean;
-  goTo: (url: string) => void;
-  agreeTermsAndPolicy: () => void;
-  onBack: () => void;
+    isOpen: boolean;
+    onClose: () => void;
+    goTo: (url: string) => void;
+    agreeTermsAndPolicy: () => void;
+    onBack: () => void;
 }
 
 function TermsModal(props: Props) {
-  const { goTo, isOpen, agreeTermsAndPolicy, onBack } = props;
-  const { t } = useTranslation();
+    const { goTo, isOpen, agreeTermsAndPolicy, onBack, onClose } = props;
+    const { t } = useTranslation();
 
-  function openTermsService() {
-    goTo('conditions/termsOfService');
-  }
-  function openPrivacyPolicy() {
-    goTo('conditions/privacyPolicy');
-  }
+    function openTermsService() {
+        goTo('conditions/termsOfService');
+    }
+    function openPrivacyPolicy() {
+        goTo('conditions/privacyPolicy');
+    }
+
+    function onModalClose(_event, reason) {
+        if (reason !== 'backdropClick') {
+            onClose();
+        }
+    }
 
-  return (
-    <Modal open={isOpen} disableBackdropClick={false}>
-      <ModalContent>
-        <Container>
-          <Title>{t('components.termsModal.welcome_to', { name })}</Title>
-          <MainContainer>
-            <TermsLogo src={termsLogoIcon} />
-            <Description>
-              <Trans i18nKey="components.termsModal.description">
-                Before we get started, please read our
-                <Link onClick={openTermsService}> Terms of Service </Link>
-                and
-                <Link onClick={openPrivacyPolicy}> Privacy Policy</Link>
-              </Trans>
-            </Description>
-          </MainContainer>
-          <BottomContainer>
-            <BackButton label={t('general.back')} onClick={() => onBack()} />
-            <AgreeButton buttonTheme="primary" onClick={() => agreeTermsAndPolicy()}>
-              {t('components.termsModal.i_agree')}
-            </AgreeButton>
-          </BottomContainer>
-        </Container>
-      </ModalContent>
-    </Modal>
-  );
+    return (
+        <Modal open={isOpen} onClose={onModalClose}>
```

Comment:
users should not be able to skip this step during the first run

Reply:
I think that the `TermsModal` and the `LanguageSelectModal` components are not used at all anywhere. I haven't found any component that imports any of these components.
The `disableBackdropClick` was removed in MaterialUI v5, so I had to replace it the `onClose={}` property. But thanks to the `reason` param, we can find what triggers closing the modal, ie. `backdrop click`:

```typescript
// Before:
<Modal open={isOpen} disableBackdropClick={false}>

// After: - here we use the "reason" to prevent closing the modal when it was triggered with the "backdropClick" event:
function onModalClose(_event, reason) {
      if (reason !== 'backdropClick') {
          onClose();
      }
  }

<Modal open={isOpen} onClose={onModalClose}>
```